lektor/reporter.py

```diff
@@ -97,6 +99,14 @@ def build(self, activity, builder):
             self.builder_stack.pop()
             self.finish_build(activity, now)
 
+    @contextmanager
```

Comment:
Sorry for the wishy-washing, but I think the API would be cleaner if the Reporter just kept a list of generic callbacks, rather than a list of Queues. The queue is an implementation detail about how the caller wants to deal with notifications (i.e. pass them via queue to another thread) — the callback should deal with that part.  If refactored this way, the API would also support clients who don't need the queue (notification processed in same thread) or clients who wish the pass the notification on to a subprocess, etc.

E.g. usage of propose API:

```py
built_artifacts = queue.Queue()

# the lambda is not really necessary, but added for clarity of intent
with reporter.notify_on_artifact_built(lambda artifact: built_artifacts.put(artifact)):
    while True:
        artifact = built_artifacts.get()
        # Do something with the artifact
```

I'm happy to work on this (after the weekend) if you don't want to.